gnosis/dex-react
# 42/return receipts

Part of #42 

This PR allows to be notified on every write operation when the transaction is delivered and when it's mined.

It's done by adding one optional param to every write operation:
- `TxOptionalParams`: Optional parameters for the transaction. It can have any relevant info, like `gasPrice`.
For now, it allows to specify a `onSendTransaction` callback that will notify when the tx has been sent to the mempool

Additionally, the write operations return now a `TxResult` with the following definition:
```js
export interface TxResult<T> {
  data: T
  receipt: Receipt
}
```

This `TxResult` will give the invoker of the write operation additional information, like the transaction hash.

All mock implementations have been updated to this new interface. Additionally the `DepositWidget` has been updated to write in the log the transaction when it's sent and when it's mined (@alfetopito placeholder for your message lib :P ). 

**Test**
1. Check: https://pr67--dexreact.review.gnosisdev.com
2. Approve a token
3. Check the console output

![image](https://user-images.githubusercontent.com/2352112/65338141-dd58bb00-dbc9-11e9-9cd1-75b0f90ca6f4.png)
